secscanner2junit/vulnerability.py

```diff
@@ -0,0 +1,189 @@
+import json
+from random import randrange
+
+from secscanner2junit.logger import warn
+
+
+class _Base:
+    def __init__(self, raw_object: dict):
+        self._raw_object = raw_object
+
+    def _parse_simple_property(self, property_name):
+        try:
+            return self._raw_object[property_name]
+        except KeyError:
+            return ''
+
+    def _parse_required_property(self, property_name):
+        try:
+            return self._raw_object[property_name]
+        except KeyError:
+            warn("Could not find required property: " + property_name + " in: " + str(self._raw_object))
+            return ''
+
+    def _parse_simple_property_list(self, property_name, property_type):
+        try:
+            elements = self._raw_object[property_name]
+            result = []
+            for element in elements:
+                result.append(property_type(element))
+            return result
+        except KeyError:
+            return []
+
+    def _parse_required_property_list(self, property_name, property_type):
+        try:
+            elements = self._raw_object[property_name]
+            result = []
+            for element in elements:
+                result.append(property_type(element))
+            return result
+        except KeyError:
+            warn("Could not find required property: " + property_name + " in: " + str(self._raw_object))
+            return []
+
+
+# https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/blob/master/src/security-report-format.json
+class _Vulnerability(_Base):
+    def __init__(self, raw_vulnerability: dict):
+        super().__init__(raw_vulnerability)
+        self.__id = self.__parse_id()
+        self.__name = super()._parse_simple_property('name')
+        self.__description = super()._parse_simple_property('description')
+        self.__severity = super()._parse_simple_property('severity')
+        self.__solution = super()._parse_simple_property('solution')
+        self.__identifiers = super()._parse_required_property_list('identifiers', Identifier)
+        self.__links = super()._parse_simple_property_list('links', Link)
+        # details
+        # tracking
+        # flags
```

Comment:
What are these about?

Reply:
there are properties that are available on schema, but not included in python class